packages/ubuntu_desktop_installer/lib/pages/allocate_disk_space/storage_columns.dart

```diff
@@ -0,0 +1,142 @@
+}
+
+class StorageDeviceColumn extends StorageColumn {
+  StorageDeviceColumn()
+      : super(
+          titleBuilder: (context) {
+            final lang = AppLocalizations.of(context);
+            return Text(lang.diskHeadersDevice);
+          },
+          diskBuilder: (context, disk) {
+            return Row(
+              children: [
+                const Icon(YaruIcons.drive_harddisk_filled),
+                const SizedBox(width: 16),
+                Text(disk.path ?? disk.id ?? ''),
+              ],
+            );
+          },
+          partitionBuilder: (context, disk, partition) {
+            return Row(
+              children: [
+                const Icon(YaruIcons.drive_harddisk),
+                const SizedBox(width: 16),
+                Text('${disk.path}${partition.number}'),
+              ],
+            );
+          },
+        );
+}
+
+class StorageTypeColumn extends StorageColumn {
+  StorageTypeColumn()
+      : super(
+          titleBuilder: (context) {
+            final lang = AppLocalizations.of(context);
+            return Text(lang.diskHeadersType);
+          },
+          diskBuilder: (context, disk) {
+            return const SizedBox.shrink();
+          },
+          partitionBuilder: (context, disk, partition) {
+            return Text(partition.format ?? '');
+          },
+        );
+}
+
+class StorageMountColumn extends StorageColumn {
+  StorageMountColumn()
+      : super(
+          titleBuilder: (context) {
+            final lang = AppLocalizations.of(context);
+            return Text(lang.diskHeadersMountPoint);
+          },
+          diskBuilder: (context, disk) {
+            return const SizedBox.shrink();
+          },
+          partitionBuilder: (context, disk, partition) {
+            return Text(partition.mount ?? '');
+          },
+        );
+}
+
+class StorageSizeColumn extends StorageColumn {
+  StorageSizeColumn()
+      : super(
+          titleBuilder: (context) {
+            final lang = AppLocalizations.of(context);
+            return Text(lang.diskHeadersSize);
+          },
+          diskBuilder: (context, disk) {
+            return Text(filesize(disk.size ?? 0));
+          },
+          partitionBuilder: (context, disk, partition) {
+            return Text(filesize(partition.size ?? 0));
+          },
+        );
+}
+
+class StorageSystemColumn extends StorageColumn {
+  StorageSystemColumn()
+      : super(
+          titleBuilder: (context) {
+            final lang = AppLocalizations.of(context);
+            return Text(lang.diskHeadersSystem);
+          },
+          diskBuilder: (context, disk) {
+            return Text(disk.ptable ?? '');
+          },
+          partitionBuilder: (context, disk, partition) {
+            return Text(partition.grubDevice == true ? 'boot' : '');
+          },
+        );
+}
+
+class StorageWipeColumn extends StorageColumn {
+  StorageWipeColumn({
+    required this.canWipe,
+    required this.isWiped,
```

Comment:
The `canWipe` and `isWiped` arguments could easily be removed, since they trivially map to properties of the `partition` object that is passed to the `partitionBuilder` method.